File: web/src/plugin/zqmcp/api/mcp_tool.js
import service from '@/utils/request'
// @Tags McpTool
// @Summary 创建MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.McpTool true "创建MCP工具管理"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /mcpTool/createMcpTool [post]
export const createMcpTool = (data) => {
  return service({
    url: '/mcpTool/createMcpTool',
    method: 'post',
    data
  })
}

// @Tags McpTool
// @Summary 删除MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.McpTool true "删除MCP工具管理"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /mcpTool/deleteMcpTool [delete]
export const deleteMcpTool = (params) => {
  return service({
    url: '/mcpTool/deleteMcpTool',
    method: 'delete',
    params
  })
}

// @Tags McpTool
// @Summary 批量删除MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除MCP工具管理"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /mcpTool/deleteMcpTool [delete]
export const deleteMcpToolByIds = (params) => {
  return service({
    url: '/mcpTool/deleteMcpToolByIds',
    method: 'delete',
    params
  })
}

// @Tags McpTool
// @Summary 更新MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.McpTool true "更新MCP工具管理"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /mcpTool/updateMcpTool [put]
export const updateMcpTool = (data) => {
  return service({
    url: '/mcpTool/updateMcpTool',
    method: 'put',
    data
  })
}

// @Tags McpTool
// @Summary 用id查询MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query model.McpTool true "用id查询MCP工具管理"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /mcpTool/findMcpTool [get]
export const findMcpTool = (params) => {
  return service({
    url: '/mcpTool/findMcpTool',
    method: 'get',
    params
  })
}

// @Tags McpTool
// @Summary 分页获取MCP工具管理列表
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取MCP工具管理列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /mcpTool/getMcpToolList [get]
export const getMcpToolList = (params) => {
  return service({
    url: '/mcpTool/getMcpToolList',
    method: 'get',
    params
  })
}
// @Tags McpTool
// @Summary 获取数据源
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /mcpTool/findMcpToolDataSource [get]
export const getMcpToolDataSource = () => {
  return service({
    url: '/mcpTool/getMcpToolDataSource',
    method: 'get',
  })
}
// @Tags McpTool
// @Summary 不需要鉴权的MCP工具管理接口
// @Accept application/json
// @Produce application/json
// @Param data query request.McpToolSearch true "分页获取MCP工具管理列表"
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /mcpTool/getMcpToolPublic [get]
export const getMcpToolPublic = () => {
  return service({
    url: '/mcpTool/getMcpToolPublic',
    method: 'get',
  })
}

File: server/plugin/zqmcp/api/mcpTool.go
package api

import (

	"github.com/flipped-aurora/gin-vue-admin/server/global"
    "github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
    "github.com/flipped-aurora/gin-vue-admin/server/plugin/zqmcp/model"
    "github.com/flipped-aurora/gin-vue-admin/server/plugin/zqmcp/model/request"
    "github.com/gin-gonic/gin"
    "go.uber.org/zap"
    "github.com/flipped-aurora/gin-vue-admin/server/utils"
)

var McpTool = new(mcpTool)

type mcpTool struct {}

// CreateMcpTool 创建MCP工具管理
// @Tags McpTool
// @Summary 创建MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.McpTool true "创建MCP工具管理"
// @Success 200 {object} response.Response{msg=string} "创建成功"
// @Router /mcpTool/createMcpTool [post]
func (a *mcpTool) CreateMcpTool(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	var info model.McpTool
	err := c.ShouldBindJSON(&info)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
    info.CreatedBy = utils.GetUserID(c)
	err = serviceMcpTool.CreateMcpTool(ctx,&info)
	if err != nil {
        global.GVA_LOG.Error("创建失败!", zap.Error(err))
		response.FailWithMessage("创建失败:" + err.Error(), c)
		return
	}
    response.OkWithMessage("创建成功", c)
}

// DeleteMcpTool 删除MCP工具管理
// @Tags McpTool
// @Summary 删除MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.McpTool true "删除MCP工具管理"
// @Success 200 {object} response.Response{msg=string} "删除成功"
// @Router /mcpTool/deleteMcpTool [delete]
func (a *mcpTool) DeleteMcpTool(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	ID := c.Query("ID")
    userID := utils.GetUserID(c)
	err := serviceMcpTool.DeleteMcpTool(ctx,ID,userID)
	if err != nil {
        global.GVA_LOG.Error("删除失败!", zap.Error(err))
		response.FailWithMessage("删除失败:" + err.Error(), c)
		return
	}
    response.OkWithMessage("删除成功", c)
}

// DeleteMcpToolByIds 批量删除MCP工具管理
// @Tags McpTool
// @Summary 批量删除MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Success 200 {object} response.Response{msg=string} "批量删除成功"
// @Router /mcpTool/deleteMcpToolByIds [delete]
func (a *mcpTool) DeleteMcpToolByIds(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	IDs := c.QueryArray("IDs[]")
    userID := utils.GetUserID(c)
	err := serviceMcpTool.DeleteMcpToolByIds(ctx,IDs,userID)
	if err != nil {
        global.GVA_LOG.Error("批量删除失败!", zap.Error(err))
		response.FailWithMessage("批量删除失败:" + err.Error(), c)
		return
	}
    response.OkWithMessage("批量删除成功", c)
}

// UpdateMcpTool 更新MCP工具管理
// @Tags McpTool
// @Summary 更新MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data body model.McpTool true "更新MCP工具管理"
// @Success 200 {object} response.Response{msg=string} "更新成功"
// @Router /mcpTool/updateMcpTool [put]
func (a *mcpTool) UpdateMcpTool(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	var info model.McpTool
	err := c.ShouldBindJSON(&info)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
    info.UpdatedBy = utils.GetUserID(c)
	err = serviceMcpTool.UpdateMcpTool(ctx,info)
    if err != nil {
        global.GVA_LOG.Error("更新失败!", zap.Error(err))
		response.FailWithMessage("更新失败:" + err.Error(), c)
		return
	}
    response.OkWithMessage("更新成功", c)
}

// FindMcpTool 用id查询MCP工具管理
// @Tags McpTool
// @Summary 用id查询MCP工具管理
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param ID query uint true "用id查询MCP工具管理"
// @Success 200 {object} response.Response{data=model.McpTool,msg=string} "查询成功"
// @Router /mcpTool/findMcpTool [get]
func (a *mcpTool) FindMcpTool(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	ID := c.Query("ID")
	remcpTool, err := serviceMcpTool.GetMcpTool(ctx,ID)
	if err != nil {
        global.GVA_LOG.Error("查询失败!", zap.Error(err))
		response.FailWithMessage("查询失败:" + err.Error(), c)
		return
	}
    response.OkWithData(remcpTool, c)
}
// GetMcpToolList 分页获取MCP工具管理列表
// @Tags McpTool
// @Summary 分页获取MCP工具管理列表
// @Security ApiKeyAuth
// @Accept application/json
// @Produce application/json
// @Param data query request.McpToolSearch true "分页获取MCP工具管理列表"
// @Success 200 {object} response.Response{data=response.PageResult,msg=string} "获取成功"
// @Router /mcpTool/getMcpToolList [get]
func (a *mcpTool) GetMcpToolList(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

	var pageInfo request.McpToolSearch
	err := c.ShouldBindQuery(&pageInfo)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	list, total, err := serviceMcpTool.GetMcpToolInfoList(ctx,pageInfo)
	if err != nil {
	    global.GVA_LOG.Error("获取失败!", zap.Error(err))
        response.FailWithMessage("获取失败:" + err.Error(), c)
        return
    }
    response.OkWithDetailed(response.PageResult{
        List:     list,
        Total:    total,
        Page:     pageInfo.Page,
        PageSize: pageInfo.PageSize,
    }, "获取成功", c)
}
// GetMcpToolDataSource 获取McpTool的数据源
// @Tags McpTool
// @Summary 获取McpTool的数据源
// @Accept application/json
// @Produce application/json
// @Success 200 {object} response.Response{data=object,msg=string} "查询成功"
// @Router /mcpTool/getMcpToolDataSource [get]
func (a *mcpTool) GetMcpToolDataSource(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

    // 此接口为获取数据源定义的数据
   dataSource, err := serviceMcpTool.GetMcpToolDataSource(ctx)
   if err != nil {
		global.GVA_LOG.Error("查询失败!", zap.Error(err))
        response.FailWithMessage("查询失败:" + err.Error(), c)
		return
   }
    response.OkWithData(dataSource, c)
}
// GetMcpToolPublic 不需要鉴权的MCP工具管理接口
// @Tags McpTool
// @Summary 不需要鉴权的MCP工具管理接口
// @Accept application/json
// @Produce application/json
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /mcpTool/getMcpToolPublic [get]
func (a *mcpTool) GetMcpToolPublic(c *gin.Context) {
    // 创建业务用Context
    ctx := c.Request.Context()

    // 此接口不需要鉴权 示例为返回了一个固定的消息接口，一般本接口用于C端服务，需要自己实现业务逻辑
    serviceMcpTool.GetMcpToolPublic(ctx)
    response.OkWithDetailed(gin.H{"info": "不需要鉴权的MCP工具管理接口信息"}, "获取成功", c)
}
